mailet/api/src/main/java/org/apache/mailet/MailetContext.java

```diff
@@ -159,17 +170,48 @@
      *
      * @param name  the attribute name
      * @param value the attribute value
+     * @deprecated see {@link #setAttribute(Attribute)
      */
+    @Deprecated
     void setAttribute(String name, Object value);
 
+    /**
+     * Associates an attribute with the given name and value with this Mail instance.
+     * If an attribute with a given name already exists, it is replaced, and the
+     * previous value is returned.
+     * <p>
+     * Conventionally, attribute names should follow the namespacing guidelines
+     * for Java packages.
+     * The Mailet API specification reserves names matching
+     * <i>org.apache.james.*</i> and <i>org.apache.mailet.*</i>.
+     *
+     * @return the previously existing attribute with the same name,
+     *      or null if there was no such attribute (or if the attribute existed
+     *      and its value was null)
+     * @since Mailet API v3.4
+     */
+    void setAttribute(Attribute attribute);
```

Comment:
this is a void method, but the documentation keeps saying `returns previous value` ?